packages/apps/reputation-oracle/server/src/common/utils/index.ts

```diff
@@ -0,0 +1,161 @@
+import { StorageCredentials, StorageParams, UploadFile } from '@human-protocol/sdk';
+import * as Minio from 'minio';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+import { PassThrough, Readable } from 'stream';
+import axios from 'axios';
+
+/**
+   * **Copy file from a URL to cloud storage**
+   *
+   * @param {string} url - URL of the source file
+   * @param {string} destBucket - Destination bucket name
+   * @param {StorageParams} clientParams - Configuration parameters for cloud storage
+   * @param {StorageCredentials} [credentials] - Optional. Cloud storage access data. If credentials are not provided, use anonymous access to the bucket
+   * @returns {Promise<UploadFile>} - Uploaded file with key/hash
+   */
+export async function copyFileFromURLToBucket(
+    url: string,
+    destBucket: string,
+    clientParams: StorageParams,
+    credentials?: StorageCredentials,
+): Promise<UploadFile> {
+    try {
+        const client: Minio.Client = new Minio.Client({
+            ...clientParams,
+            accessKey: credentials?.accessKey ?? '',
+            secretKey: credentials?.secretKey ?? '',
+        });
+
+        const { data } = await axios.get(url, { responseType: 'stream' });
+
+        const hash = await hashStream(data);
+        const key = `s3${hash}.zip`;
+
+        const stream = new PassThrough();
+        data.pipe(stream);
+
+        await new Promise((resolve, reject) => {
+            client.putObject(destBucket, key, stream, (err, etag) => {
+                if (err) reject(err);
+                else resolve(etag);
+            });
+        });
+
+        console.log(`File from ${url} copied to ${destBucket}/${key}`);
```

Comment:
Remove console.log, and use logger instead.

Reply:
Agree